internal/ota/ota.go

```diff
@@ -0,0 +1,350 @@
+package ota
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"slices"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+// UpdateReleaseAPIEndpoint updates the release API endpoint
+func (s *State) UpdateReleaseAPIEndpoint(endpoint string) {
+	s.releaseAPIEndpoint = endpoint
+}
+
+// GetReleaseAPIEndpoint returns the release API endpoint
+func (s *State) GetReleaseAPIEndpoint() string {
+	return s.releaseAPIEndpoint
+}
+
+// getUpdateURL returns the update URL for the given parameters
+func (s *State) getUpdateURL(params UpdateParams) (string, error, bool) {
+	updateURL, err := url.Parse(s.releaseAPIEndpoint)
+	if err != nil {
+		return "", fmt.Errorf("error parsing update metadata URL: %w", err), false
+	}
+
+	isCustomVersion := false
+
+	appTargetVersion := s.GetTargetVersion("app")
```

Comment:
This component code structure with an arbitrary amount of updatable components is far too complex for our needs - and would fall under pre-optimization, especially when we have hardcoded `AppTargetVersion` in the response. 

If we ever do it, adding more updateable components will be a very deliberate act, and the updating mechanism will be only a small part of that effort.